# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546904970 28800
#      Mon Jan 07 15:49:30 2019 -0800
# Branch stable
# Node ID 72e0a106d9569f5022bd497c8a43704957d115ed
# Parent  6aa84dc27d01ee47915adcfa53a42f0cd983d438
__magick_read__.cc: Fix static analyzer detected issues (bug #55347).

* __magick_read__.cc (maybe_initialize_magick): Check "locale" C string is not
null before using it in constructor of C++ string.

diff --git a/libinterp/corefcn/__magick_read__.cc b/libinterp/corefcn/__magick_read__.cc
--- a/libinterp/corefcn/__magick_read__.cc
+++ b/libinterp/corefcn/__magick_read__.cc
@@ -764,17 +764,17 @@ maybe_initialize_magick (void)
 {
   static bool initialized = false;
 
   if (! initialized)
     {
       // Save locale as GraphicsMagick might change this (fixed in
       // GraphicsMagick since version 1.3.13 released on December 24, 2011)
       const char *static_locale = setlocale (LC_ALL, nullptr);
-      const std::string locale (static_locale);
+      const std::string locale = (static_locale ? static_locale : "");
 
       const std::string program_name
         = octave::sys::env::get_program_invocation_name ();
       Magick::InitializeMagick (program_name.c_str ());
 
       // Restore locale from before GraphicsMagick initialisation
       setlocale (LC_ALL, locale.c_str ());
 
